### PR TITLE
Feature/data infra deployment doc update

### PR DESCRIPTION
**Pull request by satenderrathee**
#### 📲 What

Data platform infra Documentation update

#### 🤔 Why
		
Data Platform
		
#### 🛠 How
		
Docs updates

#### 👀 Evidence
		
Docs updated, tested on local npm server
		 
#### 🕵️ How to test

Notes on how a reviewer can test the changes, e.g. how to run the tests.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ This aligns to the workflow shown in the [deployment architecture](../architectu
 It assumes the [Azure requirements](../requirements_data_azure.md#azure) are in place, including:
 
 * Azure subscription and service principal
+* If you want to provision infrastructure within a private network, then the Azure spoke virtual network and subnet for private endpoints must be provisioned for each environment. The hub network must contain a self-hosted agent. Please refer [Hub-Spoke network topology](https://learn.microsoft.com/en-us/azure/architecture/reference-architectures/hybrid-networking/hub-spoke?tabs=cli) for more details on Hub-spoke network topology in Azure.
 * Azure DevOps project with [Pipelines variable groups](../requirements_data_azure.md#azure-pipelines-variable-groups)
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md:22]: Think you need to mention here that you need the network variable groups only if  you're provisioning in the private network (assuming that's right) - and also mention that on that page.

File: docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ This aligns to the workflow shown in the [deployment architecture](../architectu
 It assumes the [Azure requirements](../requirements_data_azure.md#azure) are in place, including:
 
 * Azure subscription and service principal
+* If you want to provision infrastructure within a private network, then the Azure spoke virtual network and subnet for private endpoints must be provisioned for each environment. The hub network must contain a self-hosted agent. Please refer [Hub-Spoke network topology](https://learn.microsoft.com/en-us/azure/architecture/reference-architectures/hybrid-networking/hub-spoke?tabs=cli) for more details on Hub-spoke network topology in Azure.
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md:21]: Does it have to use a hub-and-spoke topology, or is that just what we recommend?

File: docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md (hunk 1 above)
Comment (satenderrathee) [reply docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md:22]: That's correct its only required if we are provisioning in the private network,so we need to mention this at both pages or only in Variable group page?

Comment (adurkan-amido) [reply docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md:22]: I'd mention on both

Comment (satenderrathee) [reply docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md]: Okay updated at both location.

File: docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md (hunk 2 above)
Comment (satenderrathee) [reply docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md:21]: yes, we have to use hub-and-spoke topology only.

File: docs/workloads/azure/data/requirements_data_azure.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ the variables fall into one of two categories based on the time of requirement:
 denoting variables required at the very outset of the project, and 'After Core Infrastructure
 Deployment', referring to variables required after the fundamental infrastructure has been deployed.
 
+Please note that the variable group named `amido-stacks-euw-de-env-network` is only required if you want to provision infrastructure within a private network.
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/requirements_data_azure.md:43]:
```suggestion
ℹ️ The variables under `amido-stacks-euw-de-env-network` are only required if you want to provision the infrastructure within a private network.
```

File: docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md [hunk 4]
```diff
@@ -18,7 +18,8 @@ This aligns to the workflow shown in the [deployment architecture](../architectu
 It assumes the [Azure requirements](../requirements_data_azure.md#azure) are in place, including:
 
 * Azure subscription and service principal
-* Azure DevOps project with [Pipelines variable groups](../requirements_data_azure.md#azure-pipelines-variable-groups)
+* If you want to provision infrastructure within a private network, then the Azure spoke virtual network and subnet for private endpoints must be provisioned for each environment. The hub network must contain a self-hosted agent. Please refer [Hub-Spoke network topology](https://learn.microsoft.com/en-us/azure/architecture/reference-architectures/hybrid-networking/hub-spoke?tabs=cli) for more details on Hub-spoke network topology in Azure.
+* Azure DevOps project with [Pipelines variable groups](../requirements_data_azure.md#azure-pipelines-variable-groups). The variable group named `amido-stacks-euw-de-env-network` is only required if you want to provision infrastructure within a private network.
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/getting_started/core_data_platform_deployment_azure.md:22]: After re-reading think it's fine like this
```suggestion
* Azure DevOps project with [Pipelines variable groups](../requirements_data_azure.md#azure-pipelines-variable-groups).
```

**Review by adurkan-amido — APPROVED** [review]

**Pull request merged**